transport/src/main/java/io/netty/channel/CoalescingBufferQueue.java

```diff
@@ -37,11 +37,16 @@
 public final class CoalescingBufferQueue {
 
     private final Channel channel;
-    private final ArrayDeque<Object> bufAndListenerPairs = new ArrayDeque<Object>();
+    private final ArrayDeque<Object> bufAndListenerPairs;
     private int readableBytes;
 
     public CoalescingBufferQueue(Channel channel) {
+        this(channel, 16);
```

Comment:
nit: we use 16 here, but 8 in the real code. Perhaps we should either define a constant or add a comment as to why we picked the number?